docs/frameworks/dapr.md

```diff
@@ -0,0 +1,112 @@
+
+## Get started
+
+To get started you need to add the Dapr hosting package to your app host project by installing the [Aspire.Hosting.Dapr](https://www.nuget.org/packages/Aspire.Hosting.Dapr) NuGet package.
+
+### [.NET CLI](#tab/dotnet-cli)
+
+```dotnetcli
+dotnet add package Aspire.Hosting.Dapr
+```
+
+### [PackageReference](#tab/package-reference)
+
+```xml
+<PackageReference Include="Aspire.Hosting.Dapr"
+                  Version="[SelectVersion]" />
+```
+
+---
+
+For more information, see [dotnet add package](/dotnet/core/tools/dotnet-add-package) or [Manage package dependencies in .NET applications](/dotnet/core/tools/dependencies).
+
+The Dapr resource is added to the .NET Aspire distributed application builder using the `AddDapr()` method.
+An overload of the `AddDapr()` method that accepts Dapr options is available. For most applications, the default options will suffice.
+
+:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs" range="4":::
+
+Dapr uses the Sidecar pattern to run alongside your application. The Dapr sidecar is a lightweight, portable, and stateless HTTP server that listens for incoming HTTP requests from your application. The sidecar is responsible for managing the lifecycle of your application, including service discovery, configuration, and secrets management. To add a sidecar to a .NET Aspire resource by using the `WithDaprSidecar(string appId)` method. The `appId` parameter is the unique identifier for the Dapr application.
+
+:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs" range="18-21"  highlight="21":::
+
+The `WithDaprSidecar` method offers overloads to configure your Dapr sidecar options like app ID and ports. In the following example, the Dapr sidecar is configured with specific ports for GRPC, HTTP, metrics and a specific App ID.
+
+:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs" range="6-16"  highlight="1-7,11":::
+
+Putting everything together, here's an example of a .NET Aspire app host project which includes:
+
+- A backend that declares a Dapr sidecar with specific ports and app ID.
+- A frontend that declares a Dapr sidecar with a specific app ID and default ports.
+
+:::code language="csharp" source="snippets/Dapr/Dapr.AppHost/Program.cs":::
+
+The .NET Aspire dashboard will shows the Dapr sidecar as a resource, with its status and logs.
+
+:::image type="content" source="media/aspire-dashboard-dapr-sidecar-resources.png" lightbox="media/aspire-dashboard-dapr-sidecar-resources.png" alt-text=".NET Aspire dashboard showing Dapr sidecar resources":::
+
+## Adding the Dapr SDK
+
+To use Dapr APIs from .NET Aspire resources you can use the [Dapr SDK for ASP.NET Core](https://www.nuget.org/packages/Dapr.AspNetCore/). The Dapr SDK provides a set of APIs to interact with Dapr sidecars.
```

Comment:
A note that the `Dapr.AspNetCore` library is for when wanting to use some of the Dapr integration with ASP.NET (DI integration, registration of subscriptions, etc.); non-ASP.NET apps (e.g. console apps) can just use the `Dapr.Client` library if they just want to make calls through the Dapr sidecar.

Reply:
✅